src/useDisposable.16.17.test.ts

```diff
@@ -0,0 +1,33 @@
+import { describe, it, vi, expect } from "vitest";
+import { renderHook } from "@testing-library/react";
+import * as React from "react";
+import { useDisposable } from "./useDisposable";
+import { useStrictEffect } from "./useStrictEffect";
+import { useStrictMemo } from "./useStrictMemo";
+
+vi.mock("./useStrictMemo.ts");
+vi.mock("./useStrictEffect.ts");
+
+describe("useDisposable", () => {
+  describe("in strict mode", () => {
+    it("should not call strict effect or memo", () => {
+      const factory = vi.fn().mockReturnValue(["foo", vi.fn()]);
```

Comment:
in React 16 and 17 it's actually Ok for useMemo/useState to run twice. In these cases it's more worthwhile to test that `use-disposable` does not detect StrictMode at all